File: Practica1/S1M2/vehiculo2.0/src/cliente.java
class cliente {
    gestorFiltros filterManager;

    // Definir el gestor de filtros
   public void setFilterManager(gestorFiltros filterManager){
      this.filterManager = filterManager;
   }

   // Envía la petición al gestor de filtros
   public double sendRequest(double revoluciones, estadoMotor estadoMotor){
      return filterManager.filterRequest(revoluciones, estadoMotor);
   }
}

File: Practica4/lib/Cliente.dart
import 'Producto.dart';
import 'GestorFiltros.dart';

class Cliente{
  GestorFiltros? _gestor = null;
  List<Producto> _pedido = [];

// Constructor
  Cliente();

  List<Producto> get pedido => _pedido;
  GestorFiltros? get gestor => _gestor;

/**
 * @brief Envía la petición al filter manager y devuelve la lista de productos filtrada
 * @param producto Lista de productos
 * @param valorFiltros Lista de filtros que se van a aplicar
*/
  List<Producto> enviarPeticion(List<Producto> productos, List<List<int>> valorFiltros){
    return _gestor!.filterResquest(productos, valorFiltros);
  }

  void set setGestorFiltros(GestorFiltros g){
    _gestor = g;
  }

  void aniadirProductoPedido(Producto producto) {
    _pedido.add(producto);
  }
}